### PR TITLE
Add JsonProtocolHandler to use HdfsFileFragmenter for multi-line JSON

### PR DESCRIPTION
**Pull request by ashuka24**
This commit adds a JsonProtocolHandler which will use the HdfsFileFragmenter if we detect that the multi-line JSON profile is being used.

### CODE REVIEW

File: server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonProtocolHandler.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+    private boolean useMultilineJson(RequestContext context) {
+        String identifier = context.getOption("identifier");
+
+        if (isNotEmpty(identifier)) {
+            return true;
+        }
+
+        return false;
```
Comment (denalex) [inline server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonProtocolHandler.java:46]:
```suggestion
        return isNotEmpty(context.getOption("identifier"));
```
we can make this a 1-liner or even embed into the calling function

File: server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonProtocolHandler.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public String getAccessorClassName(RequestContext context) {
+        return context.getAccessor();
+    }
+
+    @Override
+    public String getResolverClassName(RequestContext context) {
+        return context.getResolver();
+    }
+
```
Comment (denalex) [inline server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonProtocolHandler.java:37]: nit: extra empty line

File: server/pxf-service/src/main/resources/pxf-profiles-default.xml [hunk 3]
```diff
@@ -723,6 +723,7 @@ under the License.
             <accessor>org.greenplum.pxf.plugins.json.JsonAccessor</accessor>
             <resolver>org.greenplum.pxf.plugins.json.JsonResolver</resolver>
         </plugins>
+        <handler>org.greenplum.pxf.plugins.json.JsonProtocolHandler</handler>
```
Comment (denalex) [inline server/pxf-service/src/main/resources/pxf-profiles-default.xml:726]: would the same need to be done for all `*:json profiles` ?

File: server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonProtocolHandler.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import org.greenplum.pxf.api.model.RequestContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
+
+/**
+ * Implementation of ProtocolHandler for json protocol.
+ */
+public class JsonProtocolHandler implements ProtocolHandler {
```
Comment (denalex) [inline server/pxf-json/src/main/java/org/greenplum/pxf/plugins/json/JsonProtocolHandler.java:13]: technically `json` is not a protocol, but a format, so calling it `JsonProtocolHandler` is a bit imprecise. But we only have "ProtocolHandler", not "ProfileHandler". We need to address this naming across profiles, maybe in a separate effort.

**Review by denalex — APPROVED** [review]

**Review by bradfordb-vmware — APPROVED** [review]

**Review by hpandeycodeit — APPROVED** [review]

**Pull request merged**